src/components/DropDownMenu.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useMemo, useRef, useState } from 'react';
+import { Person } from '../types/Person';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  persons: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: 300;
+};
+
+export const DropDownMenu: React.FC<Props> = ({
+  persons,
+  onSelected,
+  delay = 300,
+}) => {
+  const [focused, setFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [delayQuery, setDelayQuery] = useState('');
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const delayedQuery = debounce(setDelayQuery, delay);
```

Comment:
There is a small thing that might confuse in naming:

`delayedQuery ` sounds like the query itself (string)
And vice versa - `delayQuery ` sounds like an action.

Let's use `appliedQuery` as state value name. And `applyQuery` as debounced function